Src/AutoFixture/DataAnnotations/MinAndMaxLengthAttributeRelay.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using AutoFixture.Kernel;
+
+namespace AutoFixture.DataAnnotations
+{
+    /// <summary>
+    /// Relays a request for a constrained string to a <see cref="ConstrainedStringRequest"/>.
+    /// </summary>
+    public class MinAndMaxLengthAttributeRelay : ISpecimenBuilder
+    {
+        private readonly IMemberTypeResolver memberTypeResolver;
+
+        /// <summary>
+        /// Creates a new instance of MinAndMaxLengthAttributeRelay
+        /// </summary>
+        /// <param name="memberTypeResolver">MemberTypeResolver to determine member type of requested object</param>
+        public MinAndMaxLengthAttributeRelay(IMemberTypeResolver memberTypeResolver)
+        {
+            this.memberTypeResolver = memberTypeResolver;
+        }
+
+        /// <summary>
+        /// Creates a new specimen based on a specified minimum or maximum length of characters that are allowed.
+        /// </summary>
+        /// <param name="request">The request that describes what to create.</param>
+        /// <param name="context">A container that can be used to create other specimens.</param>
+        /// <returns>
+        /// A specimen created from a <see cref="RangedNumberRequest"/> encapsulating the operand
+        /// type, the minimum and the maximum of the requested number, if possible; otherwise,
+        /// a <see cref="NoSpecimen"/> instance.
+        /// </returns>
+        public object Create(object request, ISpecimenContext context)
+        {
+            if (request == null)
+            {
+                return new NoSpecimen();
+            }
+
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var range = Range.TryGetFromAttributes(request);
+            
+            if (range == null)
+            {
+                return new NoSpecimen();
+            }
+
+            var memberType = memberTypeResolver.TryGetMemberType(request);
```

Comment:
Please update also [`RangeAttributeRelay`](https://github.com/AutoFixture/AutoFixture/blob/11066d77bcee17e4175679cfa0e9133957695671/Src/AutoFixture/DataAnnotations/RangeAttributeRelay.cs#L52) to use this interface. Use property injection pattern there as well.

Implementation might look like following:
```c#
private static Type GetMemberType(RangeAttribute rangeAttribute, object request)
{
    if (this.MemberTypeResolver.TryGetMemberType(request, out var memberType))
    {
        return memberType;
    }

    return Nullable.GetUnderlyingType(rangeAttribute.OperandType) ?? rangeAttribute.OperandType;
}
```